src/cli.rs

```diff
@@ -62,77 +67,67 @@ impl Cli {
-        let icons = config_dir
-            .load_icons(lua_state.to_inner())
+        let icons = config_dir.load_icons(lua_inner)
             .expect("The icon configuration should be valid");
-        let colors = config_dir
-            .load_colors(lua_state.to_inner())
+        let colors = config_dir.load_colors(lua_inner)
             .expect("The color configuration should be valid");
 
-        let color_choice = self
-            .color_choice
+        // Determine color choice with less nesting
+        let color_choice = self.color_choice
             .or_else(|| config.as_ref().and_then(|config| config.color_choice()))
             .unwrap_or_default();
+
+        // Build tree with method chaining
         let mut builder = tree::Builder::new(&self.path, color_choice);
-        if let Some(config) = config {
-            builder = builder.config(config);
-        }
-        if let Some(icons) = icons {
-            builder = builder.icons(icons);
-        }
-        if let Some(colors) = colors {
-            builder = builder.colors(colors);
-        }
+        
+        // Apply configurations if they exist
+        builder = config.map(|c| builder.config(c)).unwrap_or(builder);
+        builder = icons.map(|i| builder.icons(i)).unwrap_or(builder);
+        builder = colors.map(|c| builder.colors(c)).unwrap_or(builder);
+        
         if let Some(ref git) = git {
             builder = builder.git(git);
         }
+        
         if let Some(level) = self.level {
             builder = builder.max_level(level);
         }
+        
         let tree = builder.build();
 
+        // Execute in git scope
         lua_state.in_git_scope(|| tree.write_to_stdout().map_err(mlua::Error::external))?;
 
         Ok(())
     }
 
-    /// Returns `true` if the user wants to edit a file. Edit prompts should exit early
-    /// without running the main behavior.
-    #[inline]
-    fn should_edit_file(&self) -> bool {
-        self.edit_config.is_some()
-    }
-
     /// Opens an editor for the file the user specified, creating the config directory
     /// if needed.
-    fn edit_file(&self) -> crate::Result {
+    fn edit_file(&self, edit_config: EditConfig) -> crate::Result {
         let config_dir = ConfigDir::new()?;
         fs::create_dir_all(config_dir.path())?;
 
-        let edit_config = self.edit_config.expect("Should have checked if Some");
-
-        let (file_path, contents) = match edit_config {
+        let (file_path, default_contents) = match edit_config {
             EditConfig::Config => (config_dir.main_path(), config::Main::DEFAULT_MODULE),
             EditConfig::Icons => (config_dir.icons_path(), config::Icons::DEFAULT_MODULE),
             EditConfig::Colors => (config_dir.colors_path(), config::Colors::DEFAULT_MODULE),
         };
 
-        // NOTE If we can't check if it exists, we'll be safe and skip overwriting it.
+        // Create file if it doesn't exist
```

Comment:
```suggestion
        // NOTE If we can't check if it exists, we'll be safe and skip overwriting it.
```
This note explains why we're just ignoring the try exists error.